tests/derive/groups.rs

```diff
@@ -120,3 +120,47 @@ fn helpful_panic_on_duplicate_groups() {
     use clap::CommandFactory;
     Opt::command().debug_assert();
 }
+
+#[test]
+fn required_group() {
+    #[derive(Parser, Debug)]
+    struct Opt {
+        #[command(flatten)]
+        source: Source,
+        #[command(flatten)]
+        dest: Dest,
+    }
+
+    #[derive(clap::Args, Debug)]
+    #[group(required)]
```

Comment:
atm not taking a value is for specialized cases.  Generalizing it to all bools is #3620